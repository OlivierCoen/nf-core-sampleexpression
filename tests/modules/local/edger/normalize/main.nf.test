nextflow_process {

    name "Test Process EDGER_NORMALIZE"
    script "modules/local/edger/normalize/main.nf"
    process "EDGER_NORMALIZE"
    tag "edger_normalize"

    test("Should run without failures") {

        when {

            process {
                """
                input[0] = ["E-MTAB-552", file('$baseDir/tests/input/normalize/all_counts.csv')]
                input[1] = file('$baseDir/tests/input/normalize/design.csv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
