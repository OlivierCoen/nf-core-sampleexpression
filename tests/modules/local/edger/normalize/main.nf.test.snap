{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "all_counts_normalized.csv:md5,3c200b43c176dbfead8e5595c9135801"
                ],
                "1": [
                    [
                        "EDGER_NORMALIZE",
                        "R",
                        "[1] \"R version 4.3.3 (2024-02-29)\""
                    ]
                ],
                "2": [
                    [
                        "EDGER_NORMALIZE",
                        "edgeR",
                        "[1] \u20184.0.16\u2019"
                    ]
                ],
                "csv": [
                    "all_counts_normalized.csv:md5,3c200b43c176dbfead8e5595c9135801"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-02T11:08:51.107154688"
    }
}